pkg/storage/system_intake_funding_source.go

```diff
@@ -0,0 +1,73 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// UpdateSystemIntakeFundingSources clears and updates the funding sources of a system intake
+func (s *Store) UpdateSystemIntakeFundingSources(ctx context.Context, systemIntakeID uuid.UUID, fundingSources []*models.SystemIntakeFundingSource) ([]*models.SystemIntakeFundingSource, error) {
+	now := s.clock.Now()
+
+	tx := s.db.MustBegin()
+	defer tx.Rollback()
```

Comment:
Nice use of `defer` here to make sure the transaction gets rolled back if an error occurs.